### PR TITLE
Topological mesh v2

### PR DESCRIPTION
**Pull request by dlyr**
So here is the PR for "feedback". It is not functionnal at all since I have commented out lot's of compil error generating lines.

One idea is to cleanup Meshes data structure
The other one is to have all the data in TriangleMesh (as discussed, it will be renamed mesh)
To this end it TriangleMesh have a VertexData array, that could be extended, and support things like normals, texcoord, weight.
Automatic vbo intsanciation and GPU communication will come after

### CODE REVIEW

**Comment by dlyr** [comment]
So here is the PR for "feedback". It is not functionnal at all since I have commented out lot's of compil error generating lines.

One idea is to cleanup Meshes data structure
The other one is to have all the data in TriangleMesh (as discussed, it will be renamed mesh)
To this end it TriangleMesh have a VertexData array, that could be extended, and support things like normals, texcoord, weight.
Automatic vbo intsanciation and GPU communication will come after

**Comment by dlyr** [comment]
Another solution would be to have a vector of properties, where we can dynammicaly add some (a la OpenMesh)
like 
add a prop
int texCoordProp = addProperty(new Property<Verctor3>, "texCoord")
access 
property(texCoordProp)[index]
or property("texCoord")[index]
and with an automatic vertexAttrib "texCoord" to shader in "texCoord"

File: src/Core/Mesh/TopologicalTriMesh/TopologicalMesh.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#include <Core/Mesh/TopologicalTriMesh/TopologicalMesh.hpp>
+
+namespace Ra{
+namespace Core{
+
+Scalar dot(TopoVector3 a, TopoVector3 b){
+    return a.dot(b);
```
Comment (nmellado) [inline src/Core/Mesh/TopologicalTriMesh/TopologicalMesh.cpp]: Minor comment: use `const TopoVector3 &` as input.

File: src/Core/Mesh/TopologicalTriMesh/TopologicalMesh.hpp [hunk 2]
```diff
@@ -1,35 +1,130 @@
...
 
-namespace Ra {
-namespace Core {
+namespace Ra
+{
+namespace Core
+{
+
+///\class TopoVector3 : small extension to Vector3 for OpenMesh compatibility
+class TopoVector3 : public Ra::Core::Vector3
```
Comment (nmellado) [inline src/Core/Mesh/TopologicalTriMesh/TopologicalMesh.hpp]: If we need this, we also might need implicit cast operators (in both direction) with `Ra::Core::Vector3`

File: src/Core/Mesh/TriangleMesh.hpp [hunk 3]
```diff
@@ -11,31 +11,42 @@ namespace Ra
...
             VectorArray<Triangle> m_triangles;
 
         public:
             EIGEN_MAKE_ALIGNED_OPERATOR_NEW
         };
-    }
-}
 
+
+struct TriangleMesh: public TriangleMeshT<VertexData>{
```
Comment (nmellado) [inline src/Core/Mesh/TriangleMesh.hpp]: Would better be `using TriangleMesh = TriangleMeshT<VertexData>;`

**Comment by nmellado** [comment]
> Another solution would be to have a vector of properties, where we can dynammicaly add some (a la OpenMesh)

I like this idea, looks very convenient.
It would be interesting to have this working the same way for point-clouds, i.e. to use mechanisms compatible with but independent from OpenMesh.

**Comment by dlyr** [comment]
Replaced by #305

**Pull request closed without merging**